types/time.go

```diff
@@ -3191,6 +3236,97 @@ func fullNameMonth(t *CoreTime, input string, ctx map[string]int) (string, bool)
 	return input, false
 }
 
+func yearOfWeekX(t *CoreTime, input string, ctx map[string]int) (string, bool) {
+	v, succ := parseDigits(input, 4)
+	if !succ {
+		return input, false
+	}
+	t.setYear(uint16(v))
+	ctx["X"] = 1
+	return input[4:], true
+}
+
+func yearOfWeekx(t *CoreTime, input string, ctx map[string]int) (string, bool) {
+	v, succ := parseDigits(input, 4)
+	if !succ {
+		return input, false
+	}
+	t.setYear(uint16(v))
+	ctx["x"] = 1
+	return input[4:], true
+}
+
+func weekModeV(t *CoreTime, input string, ctx map[string]int) (string, bool) {
+	v, succ := parseDigits(input, 2)
+	// v should be in [01, 53]
+	if !succ || v <= 0 || v > 53 {
+		return input, false
+	}
+	ctx["WeekNumber"] = v
+	ctx["SundayFirst"] = 1
+	ctx["V"] = 1
+	return input[2:], true
+}
+
+func weekModeU(t *CoreTime, input string, ctx map[string]int) (string, bool) {
+	v, succ := parseDigits(input, 2)
+	// v should be in [00, 53]
+	if !succ || v < 0 || v > 53 {
+		return input, false
+	}
+	ctx["WeekNumber"] = v
+	ctx["SundayFirst"] = 1
+	ctx["U"] = 1
+	return input[2:], true
+}
+
+func weekModev(t *CoreTime, input string, ctx map[string]int) (string, bool) {
+	v, succ := parseDigits(input, 2)
+	// v should be in [01, 53]
+	if !succ || v <= 0 || v > 53 {
+		return input, false
+	}
+	ctx["WeekNumber"] = v
+	ctx["v"] = 1
+	return input[2:], true
+}
+
+func weekModeu(t *CoreTime, input string, ctx map[string]int) (string, bool) {
+	v, succ := parseDigits(input, 2)
+	// v should be in [00, 53]
+	if !succ || v < 0 || v > 53 {
+		return input, false
+	}
+	ctx["WeekNumber"] = v
+	ctx["u"] = 1
+	return input[2:], true
+}
```

Comment:
It seems that most of the codes are duplicates, can we simplify them?

Reply:
No, not duplicated.
Take a careful look, you will find the tiny different between those functions. @wjhuang2016

Reply:
I see. Can we take the difference as a parameter?  Will it hurt the performance?